doc/man3/RAND_DRBG_new.pod

```diff
@@ -0,0 +1,150 @@
+=pod
+
+=head1 NAME
+
+RAND_DRBG_new,
+RAND_DRBG_secure_new,
+RAND_DRBG_set,
+RAND_DRBG_instantiate,
+RAND_DRBG_uninstantiate,
+RAND_DRBG_free
+- initialize and cleanup a RAND_DRBG instance
+
+=head1 SYNOPSIS
+
+ #include <openssl/rand_drbg.h>
+
+
+ RAND_DRBG *RAND_DRBG_new(
+                int type, unsigned int flags, RAND_DRBG *parent);
+
+ RAND_DRBG *RAND_DRBG_secure_new(
+                int type, unsigned int flags, RAND_DRBG *parent);
+
+ int RAND_DRBG_set(
+         RAND_DRBG *drbg, int type, unsigned int flags);
+
+ int RAND_DRBG_instantiate(
+         RAND_DRBG *drbg, const unsigned char *pers, size_t perslen);
+
+ int RAND_DRBG_uninstantiate(
+         RAND_DRBG *drbg);
+
+ void RAND_DRBG_free(
+         RAND_DRBG *drbg);
+
+
+=head1 DESCRIPTION
+
+B<RAND_DRBG_new>() creates a new RAND_DRBG instance.
+
+B<RAND_DRBG_secure_new>() creates a new RAND_DRBG instance from the seucre heap
+
+B<RAND_DRBG_set>()
+
+B<RAND_DRBG_instantiate>()
+
+B<RAND_DRBG_uninstantiate>()
+
+B<RAND_DRBG_free>()
+
+
+=head1 RETURN VALUES
+
+RAND_status() returns 1 if the random generator has been seeded
+with enough data, 0 otherwise.
+
+RAND_poll() returns 1 if it generated seed data, 0 otherwise.
+
+RAND_event() returns RAND_status().
+
+The other functions do not return values.
+
+=head1 NOTES
+
+The new OpenSSL DRBG has some peculiarities which need to be taken
+into account when it is selected as the default OpenSSL random generator, i.e.,
+when RAND_get_rand_method() == RAND_OpenSSL().
+This applies in particular to the way reseeding is done by the DRBG:
+
+=over 2
+
+=item *
+
+The DRBG seeds itself automatically, pulling random input from trusted
+entropy sources.
+Automatic reseeding occurs after a predefined number of generate requests.
+The selection of the trusted entropy sources is configured at build
+time using the --with-rand-seed option.
+
+=item *
+
+The DRBG distinguishes two different types of random input:
+'entropy', which comes from a trusted source, and 'additional input',
+which can optionally be added by the user and is considered untrusted.
+
+=back
+
+Automatic seeding can be disabled using the --with-rand-seed=none option.
+
+=head2 DRBG with automatic seeding enabled
+
+Calling RAND_poll() or RAND_add() is not necessary, because the DRBG
+polls the entropy source automatically.
+However, both calls are permitted, and do reseed the RNG.
+
+RAND_add() can be used to add both kinds of random input, depending on the
+value of the B<randomness> argument:
+
+=over 4
+
+=item randomness == 0:
+
+The random bytes are mixed as additional input into the current state of
+the DRBG.
+Mixing in additional input is not considered a full reseeding, hence the
+reseed counter is not reset.
+
+
+=item randomness > 0:
+
+The random bytes are used as entropy input for a full reseeding
+(resp. reinstantiation) if the DRBG is instantiated
+(resp. uninstantiated or in an error state).
+A reseeding requires 16 bytes (128 bits) of randomness.
```

Comment:
This is not the case anymore, it's now 32. We probably at least need to return the drbg->min_entropylen.